packages/utils/telemetry-utils/src/mockLogger.ts

```diff
@@ -259,3 +263,32 @@ function matchObjects(actual: ITelemetryPropertiesExt, expected: ITelemetryPrope
 	}
 	return true;
 }
+
+/**
+ * Mock {@link ITelemetryLoggerExt} implementation.
+ *
+ * @remarks Can be created via {@link IMockLoggerExt}.
+ *
+ * @internal
+ */
+export interface IMockLoggerExt extends ITelemetryLoggerExt {
+	/**
+	 * Gets the events that have been logged so far.
+	 */
+	events(): readonly ITelemetryEventExt[];
+}
+
+/**
+ * Creates an {@link IMockLoggerExt}.
+ *
+ * @internal
+ */
+export function createMockLoggerExt(minLogLevel?: LogLevel): IMockLoggerExt {
+	const mockLogger = new MockLogger(minLogLevel);
+	const childLogger = createChildLogger({ logger: mockLogger });
```

Comment:
Found while looking through the class for review.
```suggestion
	const childLogger = mockLogger.toTelemetryLogger();
```

Reply:
I considered doing it this way, but I honestly think that method should be removed. The "ext" types conceptually extend the base types, so I find it weird (and smelly) to expose knowledge about the "ext" types from a base type implementation.

There is a lot about `MockTelemetryLogger` that I think needs to be reworked. It has a few bugs, exposes state mutability, etc.

I think I'm going to leave this one alone for now, unless you feel strongly.

Reply:
Also posted this in Teams about removing this class from our (non-internal) exports: https://teams.microsoft.com/l/message/19:50292e8934024fc19d6ca2080dd7681e@thread.skype/1718659268797?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=9ce27575-2f82-4689-abdb-bcff07e8063b&parentMessageId=1718659268797&teamName=Fluid%20Framework&channelName=Dev&createdTime=1718659268797

Reply:
Nope, go ahead 👍.

For future consideration, I'd say we only need one mock implementation of a logger class which can implement both interfaces, so I guess I don't really think of it as just "a base type implementation". The Ext types ended up being alpha right now, on their way to internal, but I'd be fine having this class implement both and also end up being internal. I'm not sure if we have partners using it right now, and if we do it should be trivial for them to copy/create their own mock base logger implementation.

Reply:
Also, a work-in-progress PR where I'm attempting to make `events` readonly.
https://github.com/microsoft/FluidFramework/pull/21470